packages/eslint-config-angular/options.js

```diff
@@ -0,0 +1,67 @@
+// This is a list of standard element attributes to ignore
+// when applying the @angular-eslint/template/i18n rule
+// that requires attributes to be tagged as localizable.
+//
+// New attributes can be added here that are for element
+// configuration and not for display, where localizing
+// the attribute value would result in broken functionality.
+//
+// In cases of attribute collisions, the rule supports
+// CSS selector style element[attribute] ignore entries
+// which can be used to scope attribute ignores to specific
+// elements.
+const ignoreAttributes = {
```

Comment:
@jattasNI we have this rule disable in nimble: https://github.com/ni/nimble/blob/4e595cc0e67a9903510ba60da17afdb1bf2643d6/packages/angular-workspace/.eslintrc.js#L109

Maybe we should pre-emptively enable it to see if the example app reveals anymore attributes that need to be marked? How confident are we in the completeness of the current list / should we do any work preemptively to shore-it up for nimble?

Reply:
I am not very confident in the completeness of the list. In my [innovation day prototype PR](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/667662?_a=files&path=/Web/Workspaces/SystemLinkShared/projects/systemlink-lib-angular/i18n-tags.js) I generated it by trial and error on the SystemLinkShared libraries so it could easily be missing APIs that are used by apps (or by nobody yet) if they're not used by a shared library. I think this PR largely uses that list, so it's probably incomplete.

I am fine with making the list comprehensive either before or after completing this PR. I suspect that adopting the package in SLE apps will reveal more APIs than adopting it in the Nimble example app since the latter doesn't exercise APIs very comprehensively. But creating a local package and locally adopting it in every app is a lot of busy work so if @rbell517 would rather wait for Renovate to do that work on a production build and then add any missing attributes as a follow up PR, that's ok by me.

Similar thought on enabling i18n on the example app. I don't object to giving it a shot but I think it'll miss a lot of new APIs.

Reply:
> But creating a local package and locally adopting it in every app is a lot of busy work so if @rbell517 would rather wait for Renovate to do that work on a production build and then add any missing attributes as a follow up PR, that's ok by me.

If @rbell517 is okay keeping an eye out on rennovate PRs and helping message / coral folks running into issues post merge that works for me 👍

> I suspect that adopting the package in SLE apps will reveal more APIs than adopting it in the Nimble example app since the latter doesn't exercise APIs very comprehensively.

Yea that's true. The test suite is intended to test the angular bindings exhaustively though. Maybe we should turn on i18n in nimble-angular's test suite 🤔

Reply:
@rbell517 if you are okay with this direction then we are ready to ship

Reply:
Offline sounds like @rbell517 has additional feedback to give so cancelling autocomplete to prevent accidental release.

Reply:
Just so there are not any misunderstandings on the impact of this change, uptaking this new version in libraries and apps will not by itself "reveal more APIs." This change means errors will no longer be generated for the listed attributes when it doesn't have an i18n attribute, but it will not generate errors when the listed attributes do have an i18n attribute. That is just a limitation of the eslint rule. It also does nothing to the standard `[attribute]="'value'"` trick to avoid the eslint rule without suppressing. The net is this is strictly a loosing of the rule, no new errors should be generated by uptaking the new version, so there shouldn't be any need to monitor Renovate builds.

What I've done instead is use some regex look for cases we worked around the i18n rule (`\[[a-zA-Z-]+\]="'.*'"`) or have added a bad i18n- attribute (`i18n-[a-zA-Z-]+`) and scrubbed the results for SystemLinkShared, TestInsights, LabManagement, and SystemsManagement. This revealed 5 instances of bad i18n attributes (all Lab Management), which is actually much better than I feared. It revealed 15 more attributes to add to the ignore list, which I've pushed to this PR. Most of these are `sl-table` and `sl-grid` related, but there are also 6 new nimble attributes. Between those 4 workspaces I think we will have close to complete coverage. Please give these a look over @jattasNI @rajsite and resolve this if you're satisfied.

Reply:
Correction, only 3 instances of bad i18n- attributes. I mis-identified one and the other was already fixed.
https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/715355

Reply:
It's strictly a loosening makes sense, that hadn't computed for me. Also justifies the patch revision 👍

Sounds like part of the discussion for the UI Working Group should be to search for and remove those workarounds and update the list. Might even be tech debt tasks to farm out but that could be a working group discussion. LGTM!